### PR TITLE
ARTEMIS-351 throw an exception if we get an IOException

### PR DESCRIPTION
**Pull request by tomjenkinson**
https://issues.apache.org/jira/browse/ARTEMIS-351

### CODE REVIEW

File: artemis-journal/src/main/java/org/apache/activemq/artemis/core/journal/impl/JournalFilesRepository.java [hunk 1]
```diff
@@ -431,13 +434,16 @@ public JournalFile openFile() throws InterruptedException {
...
 
-      while (nextFile == null) {
-         nextFile = openedFiles.poll(5, TimeUnit.SECONDS);
-         if (nextFile == null) {
-            fileFactory.onIOError(ActiveMQJournalBundle.BUNDLE.fileNotOpened(), "unable to open ",  null);
-         }
+      if (nextFile == null) {
+          // We need to reconnect the current file with the timed buffer as we were not able to roll the file forward
+          // If you don't do this you will get a NPE in TimedBuffer::checkSize where it uses the bufferobserver 
+          fileFactory.activateBuffer(journal.getCurrentFile().getFile());
```
Comment (andytaylor) [inline artemis-journal/src/main/java/org/apache/activemq/artemis/core/journal/impl/JournalFilesRepository.java]: can you use the BUNDLE here Tom, see ActiveMQJournalBundle.BUNDLE

**Comment by andytaylor** [comment]
not quite tom, 

throw ActiveMQJournalBundle.BUNDLE.someException();

**Comment by andytaylor** [comment]
perfect

**Comment by clebertsuconic** [comment]
will run tests and merge it

**Comment by clebertsuconic** [comment]
There's a checkstyle error, but I will be able to amend before commit.

(Please make sure you rebase against master after I merge this).

**Pull request closed without merging**